src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml

```diff
@@ -58,6 +58,20 @@
     </Environment>
   </Provider>
 
+  <Provider Name="StackExchange" Documentation="https://api.stackexchange.com/docs/authentication">
+    <Environment Issuer="https://api.stackexchange.com/">
+      <Configuration AuthorizationEndpoint="https://stackoverflow.com/oauth"
+                     TokenEndpoint="https://stackoverflow.com/oauth/access_token/json"
+                     UserinfoEndpoint="https://api.stackexchange.com/2.3/me" />
+    </Environment>
+
+    <Setting PropertyName="ApplicationKey" ParameterName="key" Type="String" Required="true"
```

Comment:
@martincostello this parameter is required when communicating with APIs that take an access token (which is the case for the userinfo endpoint), but for some reasons, we don't throw an exception if it's not set in the aspnet-contrib provider. Should we change that?

```json
{"error_id":405,"error_message":"`key` is required when `access_token` is passed.","error_name":"key_required"}
```